R_scripts/weatherdat.R

```diff
@@ -5,13 +5,75 @@ library(readr)
 library(dplyr)
 library(ggplot2)
 library(ggrepel)
+library(lubridate)
+library(tidyr)
+
+# Function to replace S/N with descriptive label
+read_wxdat <- function(file) {
+  subfile <- read_csv("../inventory_data/wstation_info.csv")
+  wdat <- file %>% 
+    read_csv(skip=1) %>%
+    gather(label, Value, -1:-2) %>% 
+    separate(label, sep = "SEN S/N: ", into = c("firstpart", "Sensor_SN")) %>%
+    separate(firstpart, sep = ",", into = c("Sensor_Group", "info")) %>%
+    select(- info)
+  wdat$Timestamp <- mdy_hms(wdat$`Date Time, GMT-04:00`)
+  wdat$Sensor_SN <- as.integer(gsub(")", "", wdat$Sensor_SN))
+ 
+  wdat <- left_join(wdat, subfile, by = "Sensor_SN") 
+  wdat <- separate(wdat, Sensor_Label, sep = "_", into = c("info1", "info2", "Sensor_Depth"))
+  wdat <- select(wdat, -`Date Time, GMT-04:00`, -info1, -info2)
```

Comment:
Same as above. What's happening here? A comment like "# Remove unneeded and 'junk' (from separate above) columns" would help.